### ISSUE TITLE
to_spectrum_dataset() loses background model

### ISSUE THREAD

**Issue by LukasNickel** [gammapy/gammapy#5173]
**Gammapy version**
1.1, 1.2

**Bug description**
Running `to_spectrum_dataset()` on a `Mapdataset` with a background model produces a spectrum dataset, that is constructed with just the IRF bkg and not the fitted model.

**Expected behavior**
It should take the fit into account. This is the case, when the mapdataset is first stacked / `to_masked` is run, because then the model is correctly applied.

Compare these two code pieces:

https://github.com/gammapy/gammapy/blob/86488ece61f41e92ff3212b9b870f5a9f7d68b79/gammapy/datasets/map.py#L1009-L1017

https://github.com/gammapy/gammapy/blob/86488ece61f41e92ff3212b9b870f5a9f7d68b79/gammapy/datasets/map.py#L1783-L1787

As the `background` does not know of the model, it is not applied.

From my understanding of the intended behavior, the first case is correct and the second one should be fixed.

**Comment by QRemy** [comment]
Yes  indeed`to_spectrum_dataset` should `self.npred_background`instead of `self.background`.
